dependency-resolution/index.md

```diff
@@ -59,3 +59,46 @@ The rough list is then simplified, by keeping only the newest version of any lis
 Hence, `Dx` which has the highest `semver` version is selected and that is `Dx v1.5.0`.
 
 By following this algorithm, the build list is guaranteed to include the oldest module versions available that meet the requirements, thus achieving reproducible behavior.
+
+## Node.js + npm
+
+[npm](https://docs.npmjs.com/about-npm) is a package manager for [Node.js](https://nodejs.org/en), it facilitates the installation of dependencies for use by Node.js.
+To understand what dependency is used at runtime, we need to understand how both npm and Node.js work.
+
+Node.js resolves dependencies at runtime.
+To do this it starts to look for a folder structure matching the dependency name inside a `node_modules/` directory in the directory of the file importing the dependency.
+If it is not found it will repeat this process in the parent folder. This is repeated recursively until the root directory is reached ([ref](https://nodejs.org/docs/latest-v20.x/api/modules.html#loading-from-node_modules-folders)).
+Importantly, this is also how transitive dependencies are resolved.
+
+This means that for a file at `/home/example/projects/file.js` on a Unix-like system it will look for the dependency in order, returning the first instance it finds, in the following places:
+
+```text
+/home/example/projects/node_modules/
+/home/example/node_modules/
+/home/node_modules/
+/node_modules/
+```
+
+And, if the [`--no-global-search-paths`](https://nodejs.org/docs/latest-v20.x/api/cli.html#--no-global-search-paths) option isn't used, it would also look in the following places:
+
+```text
+$NODE_PATH/node_modules
+$HOME/.node_modules
+```
+
+As a package manager for Node.js, npm installs dependencies by downloading them from the internet (a package registry or source repository) and putting them in the `node_modules/` directory.
+For re-used dependencies two things may happen.
+If a common version is supported by multiple dependencies, npm will try to put the common version directly in the `node_modules/` directory of the parent project, where it can be shared.
```

Comment:
```suggestion
If a common version is supported by multiple dependencies, npm will try to put the common version directly in the `node_modules/` directory of the root project, where it can be shared.
```

I think this is clearer in that case.